src/api/v1/community_businesses/visitors.ts

```diff
@@ -80,11 +104,91 @@ const routes: Hapi.ServerRoute[] = [
 
       const visitors = await (visits
         ? Visitors.getWithVisits(knex, communityBusiness, modelQuery)
-        : Visitors.get(knex, modelQuery));
+        : Visitors.fromCommunityBusiness(knex, communityBusiness, modelQuery));
 
       return Promise.all(visitors.map(Visitors.serialise));
     },
   },
+
+  {
+    method: 'PUT',
+    path: '/community-businesses/{organisationId}/visitors/{userId}',
+    options: {
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['user_details-child:write'],
+        },
+      },
+      validate: {
+        params: {
+          userId: id,
+          organisationId: meOrId,
+        },
+        payload: {
+          name: userName,
+          gender,
+          birthYear,
+          email,
+          phoneNumber,
+          postCode,
+          isEmailConsentGranted,
+          isSMSConsentGranted,
+          disability,
+          ethnicity,
+        },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness , assign: 'communityBusiness' },
+        { method: isChildUser, assign: 'isChild' },
+      ],
+    },
+    handler: async (request: PutUserRequest, h: Hapi.ResponseToolkit) => {
+      const {
+        server: { app: { knex } },
+        payload,
+        pre: { isChild, communityBusiness },
+        params: { userId },
+      } = request;
+
+      if (!isChild) {
+        return Boom.forbidden('Insufficient permission to access this resource');
+      }
+
+      const changeset = { ...payload };
+
+      const [user] = await Visitors.fromCommunityBusiness(
+        knex,
+        communityBusiness,
+        { where: { id: Number(userId) } }
```

Comment:
This can be parsed as a number by joi validation instead of manually in the route. If we never want this value as a string this may be a better approach.

Reply:
It is parsed by Joi, the problem is that the hapi typings expect all path parameters to be strings, so this is just to satisfy the compiler.

Reply:
Typescript = the worst